### ISSUE TITLE
Reopening: Chrome will not pre-populate your "create comment" for post with string id until browser refresh in sandbox

### ISSUE THREAD

**Issue by gregbelyea71** [marmelab/react-admin#3835]
**What you were expecting:**

reopening because decodeURIComponent does not change the behaviour in your sandbox......
Same behaviour as your sandbox example for "react-admin-advanced-recipes-create-a-record-related-to-the-current-one"... I expected to have a String ID with spaces still prepopulate the dropdown and have created date show up in comment... It used to work

**What happened instead:**

Page shows up empty form, and will not populate until you hit refresh on the browser

**Steps to reproduce:**

Changed react-admin to 2.9.7, changed your id's on first two records to strings with spaces, selected the first post, went to comments tab, selected add a new comment, nothing pre populates in Chrome until you hit refresh on browser.... I tested in Microsoft Edge and it pre-populates fine

**Related code:**
https://codesandbox.io/s/react-admin-advanced-recipes-create-a-record-related-to-the-current-one-vrusd

**Environment**

React-admin version: 2.9.7
Last version that did not exhibit the issue (if applicable): 2.1.0 (tested in your sandbox worked fine)
React version: 16.10.2
Browser: Chrome Version 77.0.3865.120 (Official Build) (64-bit)
Stack trace (in case of a JS error): No error... there are some warnings but they are also present in Microsoft edge and it still works

**Comment by Kmaschta** [comment]
I don't understand this issue.. Which issue are you reopening? Can you provide a link?

**Comment by bicstone** [comment]
How is this?

https://codesandbox.io/s/react-admin-advanced-recipes-create-a-record-related-to-the-current-one-hx18s

AddCommentButton.js
```diff
const AddCommentButton = ({ classes, record }) => (
  <Button
    className={classes.button}
    variant="raised"
    component={Link}
-   to={`/comments/create?post_id=${record.id}`}
+   to={`/comments/create?post_id=${encodeURIComponent(record.id)}`}
    label="Add a comment"
    title="Add a comment"
 >
   <ChatBubbleIcon />
 </Button>
);
```

**Comment by bicstone** [comment]
Is this related to #3199 ?

**Comment by bicstone** [comment]
How is this?

https://codesandbox.io/s/react-admin-advanced-recipes-create-a-record-related-to-the-current-one-hx18s

AddCommentButton.js
```diff
const AddCommentButton = ({ classes, record }) => (
  <Button
    className={classes.button}
    variant="raised"
    component={Link}
-   to={`/comments/create?post_id=${record.id}`}
+   to={`/comments/create?post_id=${encodeURIComponent(record.id)}`}
    label="Add a comment"
    title="Add a comment"
  >
    <ChatBubbleIcon />
  </Button>
);
```

**Comment by fzaninotto** [comment]
Kevin asks for a link to the issue you're reopening. 

Also, in the codesandbox you provided, how can we reproduce the bug?

**Comment by gregbelyea71** [comment]
Apologies, i got dragged away into some graphql work and just revisiting this now.... The steps to reproduce are same as above.... but also here

Steps to reproduce:

Changed react-admin to 2.9.7, changed your id's on first two records to strings with spaces, selected the first post, went to comments tab, selected add a new comment, nothing pre populates in Chrome until you hit refresh on browser.... I tested in Microsoft Edge and it pre-populates fine

Related code:
https://codesandbox.io/s/react-admin-advanced-recipes-create-a-record-related-to-the-current-one-vrusd

`to={`/comments/create?post_id=${encodeURIComponent(record.id)}`}`

i just tried the above suggestion by @bicstone, and that appears to work in the sandbox but you can test and see what i meant because i set it back to the way it was...

**Comment by bicstone** [comment]
It might be related to #3199

**Comment by fzaninotto** [comment]
Please, don't give us half the info expecting us to find the other half - this is an open-source project, and it's already enough work with well-described issues. 

It turns out you propose to reopen #3813.

**Comment by fzaninotto** [comment]
It is fixed in 3.0. If someone in the community feels like fixing it in 2.x, please open a PR.

**Comment by fzaninotto** [comment]
> i just tried the above suggestion by @bicstone, and that appears to work in the sandbox but you can test and see what i meant because i set it back to the way it was...

I don't understand the second part of that sentence, but it seems to me that @bicstone's suggestion works, so I'm closing this issue.
